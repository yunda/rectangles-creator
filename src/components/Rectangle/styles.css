.Rectangle {
    fill: #000;
    stroke: none;
}

.Rectangle:hover {
    stroke: #f50057;
    stroke-width: 1;
    stroke-linejoin: round;
}

@media print {
    .Rectangle {
        fill: white;
    }

    .Rectangle:hover{
        stroke: none;
    }
}
