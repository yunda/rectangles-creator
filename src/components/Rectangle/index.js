import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styles from './styles.css';

class Rectangle extends Component {
    static propTypes = {
        x: PropTypes.number,
        y: PropTypes.number,
        width: PropTypes.number,
        height: PropTypes.number,
        onClick: PropTypes.func
    }

    render() {
        const {id, ...other} = this.props;
 
        return (
            <rect data-id={id} className={styles.Rectangle} {...other} />
        );
    }
}

export default Rectangle;
