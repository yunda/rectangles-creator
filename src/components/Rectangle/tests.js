import React from 'react';
import Rectangle from './index';
import {shallow} from 'enzyme';

test('Should render Rectangle', () => {
    const wrapper = shallow(
        <Rectangle width={40} height={40} x={0} y={10} />
    );

    expect(wrapper).toMatchSnapshot();
});
