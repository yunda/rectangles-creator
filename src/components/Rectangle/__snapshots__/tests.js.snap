// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Rectangle 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Rectangle
    height={40}
    width={40}
    x={0}
    y={10}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "className": undefined,
      "data-id": undefined,
      "height": 40,
      "width": 40,
      "x": 0,
      "y": 10,
    },
    "ref": null,
    "rendered": null,
    "type": "rect",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "className": undefined,
        "data-id": undefined,
        "height": 40,
        "width": 40,
        "x": 0,
        "y": 10,
      },
      "ref": null,
      "rendered": null,
      "type": "rect",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
